### ISSUE TITLE
[Feature request] Add proxy setting

### ISSUE THREAD

**Issue by Nokia808** [gi-lom/dialect#115]
Hi. 

This many be strange request from title ! But it's usefulness is that:
1) unfortunately, there are certain situation in which even such sites like online translations are blocked ! In such cases this feature will be extremely useful ...
2) to improve privacy of users, like utilizing Google via Tor network to avoid Google anti-privacy hanger....

If this feature accepted to be added as future enhancement, then it should allow SOCKS 5 proxy & should allow user to select UDP or TCP or automatic(UDP/TCP) protocol.

Thank you.

**Comment by fushinari** [comment]
Personally, I don't think this is within the scope of the application and any proxy settings should probably be system-wide. What do you think, @rafaelmardojai ?

**Comment by Nokia808** [comment]
@fushinari 
Well, at a time it is more suitable to use system-wide solution. But at other cases it is more suitable to make only certain applications bypass the blockage .......

However, the final decision is up to the developer(s)

**Comment by fushinari** [comment]
Actually, I think I agree with you that it's a good addition. But it adds a layer of complexity that I'm not sure I want. Also, to implement it properly, we would need `py-googletrans` to also support proxy settings, but sadly, it's still marked as a TODO.

**Comment by Nokia808** [comment]
@fushinari 
It is very okay to be in "TODO" list ! Take your time.

**Comment by Venturub** [comment]
I agree with this. And would suggest the user IP to be somehow replaced by another IP who would then be the one directly presenting the translation request to Google Translator. This would further justify the use of Dialect over the Google Translator website through the browser. I do not know the feasibility of this but it would be great... thanks!

**Comment by rafaelmardojai** [comment]
> The python-requests library supports it, thats why it will work for dialect.

Just for the record, Dialect doesn't use `python-requests` but `libsoup`. And it should respect the proxy settings from GNOME Settings.

**Comment by zer0-x** [comment]
Oh

So `python-requests` will be used for Google translate only?
`gTTS` uses request as I saw [here](https://github.com/pndurette/gTTS/blob/3d6cfc9d6da6a83dbfe5e46911f8ac87f19215a7/setup.cfg#L40).

**Comment by rafaelmardojai** [comment]
> So python-requests is used for Google translate only?
gTTS uses it as I saw [here](https://github.com/pndurette/gTTS/blob/3d6cfc9d6da6a83dbfe5e46911f8ac87f19215a7/setup.cfg#L40). And I saw it [here](https://github.com/dialect-app/dialect/blob/29dcb52800dbbff7f5bd88faa914de159d6a1590/build-aux/flatpak/pypi-dependencies.json#L49) in the flatpak mainfest file.

Only Google text-to-speech (gTTS), Google translation is implemented with libsoup. 

> After some search it should work with libsoup also for other translate services: https://mail.gnome.org/archives/libsoup-list/2009-February/msg00018.html

Yeah, the env vars you pointed should work with libsoup.
